atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -208,26 +231,32 @@ boolean runOnce() {
                 });
     }
 
-    /**
-     * Check whether the table being swept was dropped. If so, stop sweeping it and move on.
-     *
-     * @return Whether the table being swept was dropped
-     */
-    boolean checkAndRepairTableDrop() {
+    private SweepOutcome determineCauseOfFailure(Exception originalException, TableToSweep tableToSweep) {
         try {
             Set<TableReference> tables = specificTableSweeper.getKvs().getAllTableNames();
-            Optional<SweepProgress> progress = specificTableSweeper.getTxManager().runTaskReadOnly(
-                    specificTableSweeper.getSweepProgressStore()::loadProgress);
-            if (!progress.isPresent() || tables.contains(progress.get().tableRef())) {
-                return false;
-            } else {
-                specificTableSweeper.getSweepProgressStore().clearProgress();
-                return true;
+
+            if (!tables.contains(tableToSweep.getTableRef())) {
+                clearSweepProgress();
+                log.info("The table being swept by the background sweeper was dropped, moving on...");
+                return SweepOutcome.TABLE_DROPPED_WHILE_SWEEPING;
             }
-        } catch (RuntimeException e) {
-            log.error("Failed to check whether the table being swept was dropped."
-                    + " Continuing under the assumption that it wasn't...", e);
-            return false;
+
+            log.warn("The background sweep job failed unexpectedly; will retry with a lower batch size...",
+                    originalException);
+            return SweepOutcome.ERROR;
+
+        } catch (RuntimeException newE) {
+            log.error("Sweep failed", originalException);
+            log.error("Failed to check whether the table being swept was dropped. Retrying...", newE);
+            return SweepOutcome.ERROR;
+        }
+    }
+
+    private void clearSweepProgress() {
+        Optional<SweepProgress> progress = specificTableSweeper.getTxManager().runTaskReadOnly(
+                specificTableSweeper.getSweepProgressStore()::loadProgress);
+        if (progress.isPresent()) {
```

Comment:
Do we need to load all these values before the clear? Note that #2552 implies we might be loading a lot of data unnecessarily, calling deleteRange even if progress is empty doesnt seem to be harmful, given that we have decided to clear it by this time.